### ISSUE TITLE
Why Fastify getRequestUrl() platform adapter use req.url property instead of req.originalUrl like Express does ?

### ISSUE THREAD

**Issue by mohamed-badaoui** [nestjs/nest#5715]
## Bug Report
It seems that exclude(<path>) applied for middeware cannot work properly with fastify adapter.

## Current behavior
When fastify adapter is used, exclude(<path>) applied for middleware does not work properly.
Exclude works fine with Express middleware

## Input Code

```export class JwtModule implements NestModule {
  configure(consumer: MiddlewareConsumer): void {
    consumer
      .apply(JwtMiddleware)
      .exclude(
        { path: '/account/signin', method: RequestMethod.POST },
      )
      .forRoutes(
         ...Module1Controllers,
         ...Module2Controllers,
      );
  }
}

```

## Possible Solution
It works fine with fastify adapter only if this line returns request.originalUrl;
https://github.com/nestjs/nest/blob/v7.5.4/packages/platform-fastify/adapters/fastify-adapter.ts#L263

like express adapter does:
https://github.com/nestjs/nest/blob/v7.5.4/packages/platform-express/adapters/express-adapter.ts#L107

## Environment
Nest version: 7.5.4

**Comment by kamilmysliwiec** [comment]
If you face any issue, please, provide a minimum reproduction repository.

**Comment by micalevisk** [comment]
@kamilmysliwiec @mohamed-badaoui @jmcdo29 
Please reopen this issue. Download the the minimum reproduction from here: https://codesandbox.io/s/jolly-haslett-ewo56?file=/README.md

---

I've changing the bellow 

https://github.com/nestjs/nest/blob/b3fb080127d2f9c8211211f6972ea92e9c2a0bb1/packages/platform-fastify/adapters/fastify-adapter.ts#L259-L260

to return `request.originalUrl` instead and it worked as expected.

**Comment by jmcdo29** [comment]
@micalevisk would you mind making it a git repo? Easier to clone and work with locally

**Comment by micalevisk** [comment]
> @micalevisk would you mind making it a git repo? Easier to clone and work with locally

here you go https://github.com/micalevisk/issue-5715-nestjs-fastify

**Comment by jmcdo29** [comment]
I can reproduce it, and it looks like the fix suggested would do the job.

**Comment by micalevisk** [comment]
> I can reproduce it, and it looks like the fix suggested would do the job. Thank you for the reproduction

ty!

but the problem of using `request.originalUrl` is that when the browser perform an `GET /favicon.ico`, express-platorm returns `/favicon.ico` while, for this solution, fastify-adapter returns `undefined` :/
see:

```
> output of ExpressAdapter#getRequestUrl: /foo
> output of ExpressAdapter#getRequestUrl: /favicon.ico


> output of FastifyAdapter#getRequestUrl: /
> the value of `request.originalUrl` in FastifyAdapter#getRequestUrl: /foo
> output of FastifyAdapter#getRequestUrl: /favicon.ico
> the value of `request.originalUrl` in FastifyAdapter#getRequestUrl: undefined
```

**Comment by mohamed-badaoui** [comment]
hi,
Personally I rewrite the method like that

```
const adapter = new FastifyAdapter()
adapter.getRequestUrl = (request: CustomRequest) => {
    return request.raw ? request.raw.url : request.originalUrl || request.url;
}
```

**Comment by micalevisk** [comment]
This bug still persist in Nest v8.0.0 :disappointed:

https://github.com/nestjs/nest/blob/8b9b0dfe7b6eef7491e4146b04d4caf1a7d25030/packages/platform-fastify/adapters/fastify-adapter.ts#L263-L265

**Comment by Zclhlmgqzc** [comment]
see #7562

**Comment by kamilmysliwiec** [comment]
Fixed in 8.0.5
